### PR TITLE
Refactoring periodic section in workflow declaration

### PR DESCRIPTION
**Pull request by anthonyolea**
Replace PR #102
Co-authored with @PaulinCharliquart 

Do not change current behaviour when using (`workflows.yml`):
```yaml
periodic:
   schedule: ...
 ```
Introduce keys:
```yaml
periodic:
   interval: 42
```
 or
```yaml
periodic:
   crontab: "42 * * * *"
 ```

### CODE REVIEW

File: director/utils.py [hunk 1]
```diff
@@ -22,22 +22,53 @@ def format_schema_errors(e):
...
-            m, h, dw, dm, my = data.split(" ")
+def build_celery_schedule(workflow, data):
+    """A celery schedule can accept seconds or crontab"""
 
-            schedule = crontab(
+    def _handle_schedule(schedule):
+        try:
+            value = float(schedule)
+        except ValueError:
+            print("here")
```
Comment (PaulinCharliquart) [inline director/utils.py:32]: to be removed @anthonyolea

Comment (anthonyolea) [reply director/utils.py:32]: Good catch 👍 
Removed.

**Review by ismailrei — APPROVED** [review]
Good for me

File: docs/docs/guides/build-workflows.md [hunk 2]
```diff
@@ -86,7 +86,7 @@ example.CHAIN:
     - B
```
Comment (ncrocfer) [inline docs/docs/guides/build-workflows.md:86]: My suggestion here:

You can use the `periodic > interval` key with a number argument (unity is the second):

```yaml
example.CHAIN:
  tasks:
    - A
    - B
    - C
  periodic:
    interval: 60
```

You can also use the `periodic > crontab` key with a string argument:

```yaml
example.CHAIN_CRONTAB:
  tasks:
    - A
    - B
    - C
  periodic:
    crontab: "0 */3 * * *"
```

In this case the format is the following (the [official documentation](https://docs.celeryproject.org/en/latest/userguide/periodic-tasks.html#crontab-schedules) of the `crontab` function gives some examples of each attribute):

```yaml
periodic:
  crontab: "minute hour day_of_week day_of_month month_of_year"
```

File: docs/docs/guides/build-workflows.md [hunk 3]
```diff
@@ -98,13 +98,16 @@ example.CHAIN_CRONTAB:
...
 
-The `periodic > schedule` key takes an integer (unity is the second) or a string argument
-([crontab](https://docs.celeryproject.org/en/latest/userguide/periodic-tasks.html#crontab-schedules)
-syntax). So in the first example, the *example.CHAIN* worflow will be executed **every 60 seconds**
-and the second one, *example.CHAIN_CRONTAB*, **every three hours**.
+The `periodic > interval` key takes a number (unity is the second) or using `periodic > crontab ` key a string argument ([crontab](https://docs.celeryproject.org/en/latest/userguide/periodic-tasks.html#crontab-schedules) syntax). So in the first example, the *example.CHAIN* workflow will be executed **every 60 seconds** and the second one, *example.CHAIN_CRONTAB*, **every three hours**.
+
+
+!!! warning
+    The `periodic > schedule` key is deprecated. It is strongly advised to migrate to `interval` or `crontab` key to set up a scheduled workflow.
```
Comment (ncrocfer) [inline docs/docs/guides/build-workflows.md:108]:
```suggestion
Older version of Celery Director used the `periodic > schedule` key which is now deprecated. It is strongly advised to migrate to `periodic > interval` or  periodic > crontab` keys to set up a scheduled workflow.
```

**Review by ncrocfer — CHANGES_REQUESTED** [review]

File: docs/docs/guides/build-workflows.md (hunk 2 above)
Comment (anthonyolea) [reply docs/docs/guides/build-workflows.md:86]: Done

File: docs/docs/guides/build-workflows.md (hunk 3 above)
Comment (anthonyolea) [reply docs/docs/guides/build-workflows.md]: Done

File: docs/docs/guides/build-workflows.md [hunk 4]
```diff
@@ -98,13 +98,22 @@ example.CHAIN_CRONTAB:
...
+periodic:
+  crontab: "minute hour day_of_month month_of_year day_of_week"
+```
+
+So in the first example, the *example.CHAIN* workflow will be executed **every 60 seconds** and the second one, *example.CHAIN_CRONTAB*, **every three hours**.
+
+!!! warning
+    Older versions of Celery Director used the `periodic > schedule` key which is now deprecated.
+
+    It is strongly advised to migrate to `periodic > interval` or `periodic > crontab` keys to set up a scheduled workflow.
```
Comment (maximecaruchet) [inline docs/docs/guides/build-workflows.md:116]: * To migrate from schedule to interval: if your schedule is in seconds just rename `schedule` to `interval`
* To migrate from schedule to crontab: if your schedule uses the cron syntax rename `schedule` to `crontab`. There is also a fix when migrating to this syntax: the old `schedule` syntax incorrectly parses the cron string as `minute hour day_of_week day_of_month month_of_year` but the new `crontab` syntax properly parses the cron string as `minute hour day_of_month month_of_year day_of_week`. So be careful to this inversion when migrating here.

Comment (anthonyolea) [reply docs/docs/guides/build-workflows.md:116]: Done

**Review by maximecaruchet — APPROVED** [review]

**Review by ncrocfer — APPROVED** [review]

**Pull request merged**